src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1515 @@
+                    (df.ce_hpv_cc_status == "stage3")
+                ) & (df.ce_hpv_cc_status == df.ce_stage_at_which_treatment_given)
+            )
+        ] = self.daly_wts['stage_1_3_treated']
+
+        # Assign daly_wt to those in stage4 cancer (who have not had palliative care)
+        disability_series_for_alive_persons.loc[
+            (df.ce_hpv_cc_status == "stage4") &
+            (pd.isnull(df.ce_date_palliative_care))
+            ] = self.daly_wts['stage4']
+
+        # Assign daly_wt to those in stage4 cancer, who have had palliative care
+        disability_series_for_alive_persons.loc[
+            (df.ce_hpv_cc_status == "stage4") &
+            (~pd.isnull(df.ce_date_palliative_care))
+            ] = self.daly_wts['stage4_palliative_care']
+
+        return disability_series_for_alive_persons
+
+
+# ---------------------------------------------------------------------------------------------------------
+#   DISEASE MODULE EVENTS
+# ---------------------------------------------------------------------------------------------------------
+
+class CervicalCancerMainPollingEvent(RegularEvent, PopulationScopeEventMixin):
+    """
+    Regular event that updates all cervical cancer properties for population:
+    * Acquisition and progression of hpv, cin, cervical cancer
+    * Symptom Development according to stage of cervical Cancer
+    * Deaths from cervical cancer for those in stage4
+    """
+
+    def __init__(self, module):
+        super().__init__(module, frequency=DateOffset(months=1))
+        # scheduled to run every 1 month: do not change as this is hard-wired into the values of all the parameters.
+
+    def apply(self, population):
+        df = population.props  # shortcut to dataframe
+        m = self.module
+        rng = m.rng
+        p = self.sim.modules['CervicalCancer'].parameters
+
+        # ------------------- SET INITIAL CE_HPV_CC_STATUS -------------------------------------------------------------------
+        # this was done here and not at outset because baseline value of hv_inf was not accessible
+
+        given_date = pd.to_datetime('2010-02-03')
+
+        if self.sim.date < given_date:
+
+            women_over_15_nhiv_idx = df.index[(df["age_years"] > 15) & (df["sex"] == 'F') & ~df["hv_inf"]]
+
+            df.loc[women_over_15_nhiv_idx, 'ce_hpv_cc_status'] = rng.choice(
+                ['none', 'hpv', 'cin1', 'cin2', 'cin3', 'stage1', 'stage2a', 'stage2b', 'stage3', 'stage4'],
+                size=len(women_over_15_nhiv_idx), p=p['init_prev_cin_hpv_cc_stage_nhiv']
+            )
+
+            women_over_15_hiv_idx = df.index[(df["age_years"] > 15) & (df["sex"] == 'F') & df["hv_inf"]]
+
+            df.loc[women_over_15_hiv_idx, 'ce_hpv_cc_status'] = rng.choice(
+                ['none', 'hpv', 'cin1', 'cin2', 'cin3', 'stage1', 'stage2a', 'stage2b', 'stage3', 'stage4'],
+                size=len(women_over_15_hiv_idx), p=p['init_prev_cin_hpv_cc_stage_hiv']
+            )
```

Comment:
I think we can do this in ` initialise population`? I'm interested to know why the value for `hv_inf` is not accessible at initialise population yet we have included Hiv in the list of dependencies